app/src/main/java/com/devindi/wallpaper/history/HistoryAdapter.kt

```diff
@@ -0,0 +1,58 @@
+package com.devindi.wallpaper.history
+
+import android.net.Uri
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import com.devindi.wallpaper.R
+import com.devindi.wallpaper.model.config.ConfigManager
+import com.devindi.wallpaper.model.history.WallpaperEntry
+import com.devindi.wallpaper.model.map.buildTileRequest
+import com.squareup.picasso.Picasso
+import kotlinx.android.synthetic.main.history_item.view.*
+import timber.log.Timber
+import java.text.SimpleDateFormat
+
+class HistoryAdapter(private val configManager: ConfigManager): RecyclerView.Adapter<HistoryAdapter.HistoryItemVh>() {
+
+    var items: List<WallpaperEntry> = emptyList()
+        set(value) {
+            field = value
+            notifyDataSetChanged()
+        }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HistoryItemVh {
+        val view = LayoutInflater.from(parent.context).inflate(R.layout.history_item, parent, false)
+        return HistoryItemVh(view)
+    }
+
+    override fun getItemCount(): Int = items.size
+
+    override fun onBindViewHolder(holder: HistoryItemVh, position: Int) {
+        val locationFormat = "%.2f %.2f"
+        val timeFormat = SimpleDateFormat()
+
+        val entry = items[position]
+        val mapSource = configManager.config.availableSources.find { it.id == entry.mapSourceId }!!
+
+        val tileRequest = buildTileRequest(mapSource.id, entry.width, entry.height, 15, entry.centerLat, entry.centerLon)
+        Timber.d("Binding history item $entry, $mapSource $tileRequest")
+        Picasso.with(holder.itemView.context)
+            .load(Uri.parse(tileRequest))
+            .into(holder.mapPreview)
+        holder.mapTitle.text = mapSource.title
+        holder.wallpaperCoordinates.text = String.format(locationFormat, entry.centerLat, entry.centerLon)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/devindi/mapwall/blob/c641b62fa39c60acdf91edee9483647734123681/app/src/main/java/com/devindi/wallpaper/history/HistoryAdapter.kt#L47"></span>Exceeded max line length (100) (cannot be auto-corrected)</td>
    </tr>
  </tbody>
</table>